### PR TITLE
Speed up imports

### PR DESCRIPTION
**Pull request by hugohadfield**
Speed up imports (especially of conformal algebras) by removing the function preloading and adding some minor caching where possible

### CODE REVIEW

File: clifford/_layout.py [hunk 1]
```diff
@@ -572,9 +560,8 @@ def _gen_complement_func(self, omt):
 
         # Since we're working with basis blades, we can use the table directly.
         # We only care about the pseudo-scalar part of the wedge.
-        omt_ps_part = omt[:, -1, :]
         for n in range(dims):
-            signlist[n] = (-1)**(omt_ps_part[n, dims-1-n] < 0.001)
+            signlist[n] = (-1)**(omt[n, -1, dims-1-n] < 0.001)
```
Comment (eric-wieser) [inline clifford/_layout.py:564]: Did you profile this? was it actually faster?

Comment (hugohadfield) [reply clifford/_layout.py:564]: After change:
image
Before change:
image
Good call! Will revert now :)

File: clifford/_layout.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def omt_check(grade_v, grade_i, grade_j):
     return grade_v == (grade_i + grade_j)
 
 
-@_numba_utils.njit
+@_numba_utils.njit(cache=True)
```
Comment (eric-wieser) [inline clifford/_layout.py:89]: Any reason for not caching `_numba_construct_gmt`?

File: clifford/_layout.py [hunk 3]
```diff
@@ -353,18 +353,6 @@ def __init__(self, *args, **kw):
...
-        self.gmt_func
-        self.imt_func
-        self.omt_func
-        self.lcmt_func
-        self.adjoint_func
-        self.left_complement_func
-        self.right_complement_func
-        self.dual_func
-        self.vee_func
-        self.inv_func
```
Comment (eric-wieser) [inline clifford/_layout.py:366]: I worry this change will have made this benchmark meaningless:

https://github.com/pygae/clifford/blob/ce0bc74520451a05ab156bd07ce3db783b4f0839/clifford/test/test_algebra_initialisation.py#L14-L22

Can you add these lines to the benchmark?

**Review by eric-wieser — COMMENTED** [review]
Note that numba/numba#6373 should allow us to cache `_numba_construct_graded_mt` too - but no need to wait for that for this PR.

**Comment by hugohadfield** [comment]
To test the import speed up thanks to this caching we did the following test:
1. Install clifford in developer mode
2. Run the following script, `time_import.py`, just to get the general python caching going etc. Don't bother recording the result
```
import numba, numpy, time, sparse 
t = time.perf_counter()
import clifford.g3c
print(time.perf_counter() - t)
```
3. In the `clifford/clifford/__pycache__` directory delete all `.nbi` and `.nbc` files
4. Run  `time_import.py` again to get the effect of jitted but uncached clifford functions
5. Run  `time_import.py` again to get  the effect of jitted and cached clifford functions

Results on my home PC:
Uncached 1.8571729790000973
Cached 0.9225598940010968

We also repeated the experiment but changing it also test the time required to also multiply two mvs:
```
import numba, numpy, time, sparse
t = time.perf_counter()
from clifford.g3c import *
print(e1*e2)
print(time.perf_counter() - t)
```
Results 
Uncached 1.9949228559999028
Cached 1.0691611509992072

**Pull request merged**